src/jupyter_contrib_nbextensions/nbextensions/select_keymap/main.js

```diff
@@ -0,0 +1,308 @@
+                }
+            },
+            remove: {
+                edit_shortcuts: ["ctrl-shift-minus"],
+                keyMap: ["Ctrl-V"]
+            }
+        }
+    };
+
+    var base_url = utils.get_body_data("baseUrl");
+    var config = new configmod.ConfigSection('notebook', {
+        base_url: base_url
+    });
+
+    config.load();
+
+    var conf = {
+        line_wrap: true
+    };
+
+    // make sure config is loaded before making changes
+    config.loaded.then(function() {
+        for (var key in conf) {
+            var config_key = "select_keymap_" + key;
+            if (config.data.hasOwnProperty(config_key)) {
+                conf[key] = config.data[config_key];
+            }
+        }
+        save_starting_state();
+        // initialize last stored value or default
+        switch_keymap(window.localStorage.getItem("jupyter_keymap") || "default");
+    });
+
+    function flatten_shortcuts(shortcut_tree) {
+        var result = {};
+        for (var p in shortcut_tree) {
+            if (typeof(shortcut_tree[p]) == "string") {
+                result[p] = shortcut_tree[p];
+            } else {
+                var subresult = flatten_shortcuts(shortcut_tree[p]);
+                for (var subp in subresult) {
+                    result[p + "," + subp] = subresult[subp];
+                }
+            }
+        }
+        return result;
+    }
+
+    function update_shortcuts(shortcut_manager, updated_shortcuts) {
+        var current_shortcuts = _.invert(flatten_shortcuts(shortcut_manager._shortcuts));
+
+        for (var shortcut_action in _.invert(updated_shortcuts)) {
+            if (_.has(current_shortcuts, shortcut_action) &&
+                shortcut_manager.get_shortcut(current_shortcuts[shortcut_action])) {
+
+                shortcut_manager.remove_shortcut(current_shortcuts[shortcut_action]);
+            }
+        }
+
+        shortcut_manager.add_shortcuts(updated_shortcuts);
+    }
+
+    // start custom CodeMirror command definitions
+    function leave_current_mode(cm) {
+        if (cm.state.vim.insertMode) {
+            // Move from insert mode into command mode.
+            CodeMirror.keyMap['vim-insert'].call('Esc', cm);
+        } else if (cm.state.vim.visualMode) {
+            // Move from visual mode to command mode.
+            CodeMirror.keyMap['vim'].call('Esc', cm);
+        } else {
+            // Move to notebook command mode.
+            Jupyter.notebook.command_mode();
+            Jupyter.notebook.focus_cell();
+        }
+    };
+
+    var orig_ctrl_y = CodeMirror.keyMap.emacs["Ctrl-Y"];
+
+    function yank_no_selection(cm) {
+        orig_ctrl_y(cm);
+        // remove selection after yank
+        cm.setSelection(cm.getCursor(), cm.getCursor());
+    };
+
+    CodeMirror.commands.yank_no_selection = yank_no_selection;
+    CodeMirror.commands.leave_current_mode = leave_current_mode;
+    // end custom CodeMirror command definitions
+
+    function create_menu() {
```

Comment:
It would be best to add a menu mirroring the current layout of the notebook `edit` page, which add keymaps as a sub-heading under the "edit" tab. For example:

````
$("#edit_menu").append('<li class="divider"></li>');
$("#edit_menu").append('<li class="dropdown-header">Key Map</li>');

available_keymaps.forEach(function(keymap) {
    $("#edit_menu").append(
        `<li><a id="keymap-${keymap}" href="#" title="Toggle ${keymap} keybindings" onClick="switch_keymap('${keymap}')" style="text-transform: capitalize;">${keymap}</a></li>`);
});